modules/dockermodelrunner/dockermodelrunner_test.go

```diff
@@ -0,0 +1,96 @@
+package dockermodelrunner_test
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/testcontainers/testcontainers-go"
+	"github.com/testcontainers/testcontainers-go/modules/dockermodelrunner"
+	"github.com/testcontainers/testcontainers-go/modules/socat"
+)
+
+const (
+	testModelNamespace       = "ai"
+	testModelName            = "llama3.2"
+	testModelTag             = "latest"
+	testModelFQMN            = testModelNamespace + "/" + testModelName + ":" + testModelTag
+	testModelNameNonExistent = "non-existent"
+	testNonExistentFQMN      = testModelNamespace + "/" + testModelNameNonExistent + ":" + testModelTag
+)
+
+func TestRun(t *testing.T) {
+	ctx := context.Background()
+
+	t.Run("success", func(t *testing.T) {
+		ctr, err := dockermodelrunner.Run(ctx)
+		testcontainers.CleanupContainer(t, ctr)
+		require.NoError(t, err)
+	})
+
+	t.Run("success/with-image", func(t *testing.T) {
+		ctr, err := dockermodelrunner.RunWithImage(ctx, socat.DefaultImage)
+		testcontainers.CleanupContainer(t, ctr)
+		require.NoError(t, err)
+	})
+
+	t.Run("failure/with-image", func(t *testing.T) {
+		_, err := dockermodelrunner.RunWithImage(ctx, "alpine:latest")
```

Comment:
bug: while its expecting an error it should handle not getting one, so call CleanupContainer.